src/pf-accordion/pf-accordion.component.js

```diff
@@ -0,0 +1,315 @@
+import PfAccordionPanel from 'pf-accordion-panel.component';
+import PfAccordionTemplate from 'pf-accordion-template.component';
+import {
+  pfUtil
+} from 'pf-utils';
+
+/**
+ * <b>&lt;pf-accordion&gt;</b> element for Patternfly Web Components
+ *
+ * @example {@lang xml}
+ * <pf-accordion>
+ *   <pf-accordion-panel>
+ *     <pf-accordion-heading>
+ *       <h4 class="panel-title">
+ *         <a role="button" data-toggle="collapse" href="#">
+ *           Collapsible Group Item #1
+ *         </a>
+ *       </h4>
+ *     </pf-accordion-heading>
+ *     <pf-accordion-template open>
+ *       <pf-accordion-body>
+ *         Collapse CONTENT 1
+ *       </pf-accordion-body>
+ *     </pf-accordion-template>
+ *   </pf-accordion-panel>
+ *   <pf-accordion-panel class="panel panel-primary">
+ *     <pf-accordion-heading>
+ *       <h4 class="panel-title">
+ *         <a role="button" data-toggle="collapse" href="#">
+ *           Collapsible Group Item #2
+ *         </a>
+ *       </h4>
+ *     </pf-accordion-heading>
+ *     <pf-accordion-template>
+ *       <pf-accordion-body>
+ *         Collapse CONTENT 2
+ *       </pf-accordion-body>
+ *     </pf-accordion-template>
+ *   </pf-accordion-panel>
+ * </pf-accordion>
+ *
+ * @prop fixedHeight {Boolean} Whether the accrodion is a fixed-height accordion
+ */
+export class PfAccordion extends HTMLElement {
+  /**
+   * Called when an instance of the element is created
+   */
+  createdCallback() {
+    this._openPanels = [];
+    this._fixedHeight = false;
+  }
+
+  /**
+   * Called when an instance was inserted into the document
+   */
+  attachedCallback() {
+    this.classList.add('panel-group');
+    this.setAttribute('role', 'tablist');
+    this.setAttribute('aria-multiselectable', 'true');
+
+    let nodes = this.querySelectorAll('pf-accordion-panel > pf-accordion-template');
+    if (nodes) {
+      Array.prototype.forEach.call(nodes, ((panel) => {
+        this._checkAndAddPanel(panel);
+      }));
+    }
+
+    // catch bubbled events
+    this.addEventListener('pf-accordion.expanding', this._handlePanelShown);
+    this.addEventListener('pf-accordion.collapsing', this._handlePanelHidden);
+
+    this._observer = new MutationObserver((mutations) => {
+      mutations.forEach((mutationRecord) => {
+        if ('childList' === mutationRecord.type) {
+
+          // handle dynamic addition of panels
+          for (let i = 0; i < mutationRecord.addedNodes.length; i++) {
```

Comment:
```addedNodes``` and ```removedNodes``` are also node lists might be good to apply the same looping syntax for all node lists